check expected modem firmware

coderbyheart opened the pull request:
This adds a one-time check which compares the modem firmware
to an expected version configured in 
`CONFIG_EXPECTED_MODEM_FIRMWARE_VERSION`.

Closes #70

coderbyheart committed:
style: fix Clang error

Signed-off-by: Markus Tacker <Markus.Tacker@NordicSemi.no>

simensrostad commented on src/main.c:
```diff
@@ -352,6 +352,30 @@ accelerometer_buffer_populate(const struct ext_sensor_evt *const acc_data)
+	}
+	if (strcmp(&modem_param->device.modem_fw.value_string,
+		   CONFIG_EXPECTED_MODEM_FIRMWARE_VERSION) != 0) {
+		LOG_ERR("Unsupported modem firmware version: %s",
```
I would use ```LOG_WRN``` here. ```LOG_ERR``` should be reserved severe errors/faults.

coderbyheart committed:
feat: check expected modem firmware

This adds a one-time check which compares the modem firmware
to an expected version configured in
`CONFIG_EXPECTED_MODEM_FIRMWARE_VERSION`.

Closes #70

Signed-off-by: Markus Tacker <Markus.Tacker@NordicSemi.no>
Co-authored-by: Simen Røstad <Simen.Rostad@NordicSemi.no>

coderbyheart committed:
style: remove redundant $ZEPHYR_BASE from sources

The $srctree environment variable is already set to point to the Zephyr
root, so no need to do

    source "$(ZEPHYR_BASE)/Kconfig.zephyr"

in samples. Just

    source "Kconfig.zephyr"

works.

Signed-off-by: Markus Tacker <Markus.Tacker@NordicSemi.no>

simensrostad commented on src/main.c:
```diff
@@ -352,6 +352,30 @@ accelerometer_buffer_populate(const struct ext_sensor_evt *const acc_data)
+			CONFIG_EXPECTED_MODEM_FIRMWARE_VERSION);
+		LOG_WRN("You can change the expected version through the");
+		LOG_WRN("EXPECTED_MODEM_FIRMWARE_VERSION setting.");
+		LOG_WRN("Please upgrade: http://bit.ly/nrf9160-mfw-update")
```
Missing ```;```

simensrostad commented on src/main.c:
```diff
@@ -352,6 +352,30 @@ accelerometer_buffer_populate(const struct ext_sensor_evt *const acc_data)
+	if (strcmp(modem_param.device.modem_fw.value_string,
+		   CONFIG_EXPECTED_MODEM_FIRMWARE_VERSION) != 0) {
+		LOG_WRN("Unsupported modem firmware version: %s",
+			modem_param.device.modem_fw.value_string);
```
missing ```log_strdup```

simensrostad commented on src/main.c:
```diff
@@ -352,6 +352,30 @@ accelerometer_buffer_populate(const struct ext_sensor_evt *const acc_data)
+		LOG_WRN("Please upgrade: http://bit.ly/nrf9160-mfw-update");
+	} else {
+		LOG_INF("Board is running expected modem firmware version: %s",
+			modem_param.device.modem_fw.value_string);
```
missing ```log_strdup```

simensrostad reviewed (approved):
LGMT
